tendermint/src/lite_impl/signed_header.rs

```diff
@@ -53,36 +54,43 @@ impl lite::Commit for block::signed_header::SignedHeader {
     fn validate(&self, vals: &Self::ValidatorSet) -> Result<(), Error> {
         if self.commit.precommits.len() != vals.validators().len() {
             fail!(
-                lite::error::Kind::ImplementationSpecific,
+                Kind::ImplementationSpecific,
                 "pre-commit length: {} doesn't match validator length: {}",
                 self.commit.precommits.len(),
                 vals.validators().len()
             );
         }
 
-        // make sure each vote is for the correct header
         for precommit_opt in self.commit.precommits.iter() {
             match precommit_opt {
                 Some(precommit) => {
+                    // make sure each vote is for the correct header
                     if let Some(header_hash) = precommit.header_hash() {
                         if header_hash != self.header_hash() {
                             fail!(
-                                lite::error::Kind::ImplementationSpecific,
+                                Kind::ImplementationSpecific,
                                 "validator({}) voted for header {}, but current header is {}",
                                 precommit.validator_address,
                                 header_hash,
                                 self.header_hash()
                             );
                         }
                     }
+
+                    // returns FaultyFullNode error if it detects a signer isn't present in the validator set
```

Comment:
I think this is actually an ImplementationSpecific error, since it depends on Commits containing the validator set addresses. So we don't need a new error type in the light client.

Reply:
Makes sense @ebuchman. Will address that :)